### ISSUE TITLE
BUG: No proper heuristic for selection of reference in EOdal mapper

### ISSUE THREAD

**Issue by lukasValentin** [EOA-team/eodal#90]
In version [0.2.1](https://github.com/EOA-team/eodal/releases/tag/v0.2.1) we announced that the grid alignment issue of scenes from different projections should be fixed in the mapper call. However, this seems still not to be completely true as shown when trying to fetch data from two Sentinel-2 tiles in two different UTM zones.

This is because of a flaw in the mapper: While we select the largest common spatial bounding box of all scenes and then force the mapper to align all scenes into that bounding box, we do not check if the grids are aligned (reprojection errors and inaccuracies!) and the pixel size matches as we can see from the [code snippet below](https://github.com/EOA-team/eodal/blob/6ff0a54d3cdd16e66ff95a2f5622a254e9b045fb/eodal/mapper/mapper.py#L717):

```python
minx, maxy = total_bounds[0], total_bounds[-1]
dst_transform = Affine(
    a=geo_info.pixres_x,
    b=0,
    c=minx,
    d=0,
    e=geo_info.pixres_y,
    f=maxy)
```
While `total_bounds` is the same for all scenes (desired), `geo_info` potentially different for the single scenes (see [here](https://github.com/EOA-team/eodal/blob/6ff0a54d3cdd16e66ff95a2f5622a254e9b045fb/eodal/mapper/mapper.py#L710)):

```python
for _, scene in scoll:
    if scene.is_bandstack():
        # get a band of the SceneCollection
        reference_band = scene[scene.band_names[0]]
        geo_info = reference_band.geo_info
```
Thus, while the total bounds are enforced correctly, the pixel size could still differ and thus the number of rows and columns per scene. As a solution, we need to select a single reference scene for all scenes in the mapper.

However, we need to define such a heuristic. I'd propose to use the following:

1. Are there scenes that have not been reprojected (least error)?
2. If NO: use the first scene in the collection as reference
3. If YES: use this scene as reference

**Comment by lukasValentin** [comment]
The heuristic has been implemented in the `mapper` class:

```python
if not self.metadata._duplicated.all():
    reference_time = self.metadata[
        ~self.metadata._duplicated][self.time_column].values[0]
    timestamps = scoll.timestamps
    reference_timestamp_idx = np.argmin(
        [pd.to_datetime(x) - reference_time for x in timestamps])
    reference_scene = scoll[
        timestamps[reference_timestamp_idx]]
else:
    reference_scene = scoll[scoll.timestamps[0]]
```
We use the `_duplicated` attribute of the metadata table to identify scenes that were not mosaiced. This is a bit sub-optimal since we could still get a scene that was resampled...

**Comment by lukasValentin** [comment]
While debugging the `mapper`, I also realized that the spatial reprojection method was always called - even it was not necessary at all (e.g., reprojection from UTM32 to UTM32 what is obviously non-sense). The consequence, however, was potential floating point inaccuracies in the pixel size (e.g., a pixel size of `10.0` became `9.9999997` or so). This has been fixed by checking if reprojection is required.
